File: Dashboard/include/regression.py
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error
import json

# Lire les données du fichier JSON
with open('data.json', 'r') as f:
    data = json.load(f)

# Convertir les données en DataFrame
df = pd.DataFrame(data)

# Séparer les variables indépendantes et la variable dépendante
X = df[['duree_formation', 'prix_formation']]
y = df['nb_inscriptions']

# Diviser les données en ensembles d'entraînement et de test
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

# Créer et entraîner le modèle de régression linéaire multiple
model = LinearRegression()
model.fit(X_train, y_train)

# Faire des prédictions sur l'ensemble de test
y_pred = model.predict(X_test)

# Évaluer le modèle
mse = mean_squared_error(y_test, y_pred)

# Stocker les résultats dans un dictionnaire
results = {
    'mse': mse,
    'coefficients': model.coef_.tolist(),
    'intercept': model.intercept_.tolist()
}

# Écrire les résultats dans un fichier JSON
with open('results.json', 'w') as f:
    json.dump(results, f)

File: Dashboard/include/regression.php
<?php
// Inclure le fichier de connexion à la base de données
include("include/connect.php");

// Requête SQL pour obtenir les données nécessaires pour la régression multiple
$sql = "SELECT duree_formation, prix_formation, COUNT(id) AS nb_inscriptions
        FROM inscription
        GROUP BY duree_formation, prix_formation";
$result = $conn->query($sql);

// Tableaux pour stocker les données
$durees = [];
$prix = [];
$inscriptions = [];

if ($result->num_rows > 0) {
    while ($row = $result->fetch_assoc()) {
        $durees[] = $row['duree_formation'];
        $prix[] = $row['prix_formation'];
        $inscriptions[] = $row['nb_inscriptions'];
    }
}

// Fermer la connexion à la base de données
$conn->close();

// Créer un tableau associatif
$data = [
    'duree_formation' => $durees,
    'prix_formation' => $prix,
    'nb_inscriptions' => $inscriptions
];

// Écrire les données dans un fichier JSON
file_put_contents('data.json', json_encode($data));
?>
